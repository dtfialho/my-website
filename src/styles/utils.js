// Colors
export const backgroundColor = '208,73,37'
export const backgroundColorDark = '153,46,36'
export const textColorLight = '255,255,255'
export const textColor = '100,100,100'
export const textColorDark = '51,51,51'

//Placeholders
export const container = `
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 15px;
`

//Mixins
export const phoneOnly = (styles) => `
  @media (max-width: 599px) { ${styles} }
`
export const tabletPortrait = (styles) => `
  @media (min-width: 600px) { ${styles} }
`
export const tabletLandscape = (styles) => `
  @media (min-width: 900px) { ${styles} }
`
export const desktop = (styles) => `
  @media (min-width: 1200px) { ${styles} }
`
export const largeDesktop = (styles) => `
  @media (min-width: 1800px) { ${styles} }
`
