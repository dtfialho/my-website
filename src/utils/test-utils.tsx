import type { ReactNode } from 'react'
import { render, RenderResult } from '@testing-library/react'

const getFileTranslations = (locale = 'pt-BR', key: string) => {
  const [file, prop] = key.split(':')
  const data = require(`locales/${locale}/${file}.json`)
  return data[prop]
}

export const renderWithTranslate = (
  children: ReactNode,
  locale = 'pt-BR'
): RenderResult => {
  const useTranslation = jest.spyOn(
    require('next-translate/useTranslation'),
    'default'
  )
  useTranslation.mockImplementation(() => ({
    t: (key: string) => getFileTranslations(locale, key)
  }))

  return render(children)
}
