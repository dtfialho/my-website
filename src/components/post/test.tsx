import { screen } from '@testing-library/react'
import format from 'date-fns/format'

import { renderWithTranslate } from 'utils/test-utils'
import Post, { PostType } from './'

const data: PostType = {
  slug: 'slug-for-test',
  date: '2022-05-27T00:00:00-03:00',
  hero_image: '/img/bg.jpg',
  title: 'Some title for testing',
  excerpt: 'lorem ipsum dolor sit amet'
}

describe('Components/Post', () => {
  it('Should render correctly', () => {
    const { container } = renderWithTranslate(<Post {...data} />)
    expect(container).toMatchSnapshot()
  })

  it('Should render correctly in en', () => {
    const { container } = renderWithTranslate(<Post {...data} />)
    expect(container).toMatchSnapshot()
  })

  it('Should render with the correct given props', () => {
    const parsedDate = format(new Date(data.date), 'dd/MM/yyyy')

    renderWithTranslate(<Post {...data} />)

    expect(screen.getByRole('link', { name: data.title })).toHaveAttribute(
      'href',
      `/blog/${data.slug}`
    )
    expect(screen.getByAltText(data.title)).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: data.title })
    ).toBeInTheDocument()
    expect(screen.getByText(data.excerpt)).toBeInTheDocument()
    expect(screen.getByText(parsedDate)).toBeInTheDocument()
  })
})
