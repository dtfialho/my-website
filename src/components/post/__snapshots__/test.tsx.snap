// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Post Should render correctly 1`] = `
.c0:not(:last-child) {
  margin-bottom: 24px;
  padding-bottom: 24px;
  border-bottom: 1px solid #d9d9d9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
}

.c1:hover {
  opacity: 0.7;
}

.c2 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  width: 100%;
  height: 150px;
  margin: 0 auto;
}

.c2 img {
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c3 {
  padding-top: 16px;
}

.c4 {
  margin-bottom: 4px;
}

.c5 {
  margin-bottom: 4px;
}

.c6 {
  font-size: 0.8rem;
  font-style: italic;
  margin-bottom: 0;
  color: #666;
}

@media (min-width:768px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:450px) and (max-width:768px) {
  .c2 {
    width: 500px;
    height: 300px;
    max-width: 100%;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 300px;
    margin: 0;
  }
}

@media (min-width:768px) {
  .c3 {
    padding: 16px 24px 0;
  }
}

<div>
  <article
    class="c0"
  >
    <a
      class="c1"
      href="/blog/slug-for-test"
      title="Some title for testing"
    >
      <figure
        class="c2"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Some title for testing"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </span>
      </figure>
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Some title for testing
        </h2>
        <p
          class="c5"
        >
          lorem ipsum dolor sit amet
        </p>
        <em
          class="c6"
        >
          Em: 
          <span>
            27/05/2022
          </span>
        </em>
      </div>
    </a>
  </article>
</div>
`;
